### ISSUE TITLE
[Feature Request] Can the Date-Picker add a value-Format configuration like the 2.x version that allows you to configure the value types (such as timestamps) of the values to which the component is bound

### ISSUE THREAD

**Issue by sufengyong** [element-plus/element-plus#1068]
### Existing Component
是

### Component Name
Date-Picker

### Description
Can the Date-Picker add a value-Format configuration like the 2.x version that allows you to configure the value types (such as timestamps) of the values to which the component is bound

**Comment by zazzaz** [comment]
Since we have Day.js installed already, you can simply use

```
import dayjs from 'dayjs'
dayjs(modelValue).format()
dayjs(modelValue).unix()
dayjs(modelValue).valueOf()
```
ref: https://day.js.org/docs/en/display/format#list-of-all-available-formats
to be more flexible than the old value-Format props.

**Comment by Pally-l** [comment]
What value do I need to pass to modify the Date object returned by the component by default.

**Comment by Pally-l** [comment]
Because the format conversion is required every time, this is redundant

**Comment by zazzaz** [comment]
It's not hard to use a computed value e.g.
```
// <el-date-picker v-model="value"></ el-date-picker>
data: {
  value: new Date()
},
computed: {
  valueFormat() {
    return dayjs(this.value).unix()
  }
},
```

**Comment by Pally-l** [comment]
If multiple time attributes are bound to an object, don't you need to merge the objects every time?

**Comment by zazzaz** [comment]
well, the answer is yes, but still, personally, I do think it is good to pass in and out via native Date Object

**Comment by npm-ued** [comment]
If i have many fields i must transform much time，it not speedy

**Comment by tracker1999** [comment]
> well, the answer is yes, but still, personally, I do think it is good to pass in and out via native Date Object

I have many fields
I must transform much time
Especially when type is range

**Comment by recursion123** [comment]
@tracker1999 In my project, I deal with it like this now. but i think it's not a good solution，beacuse I don't want to override every attribute of el-date-picker
```
<template>
  <el-date-picker type="date" :placeholder="placeholder" :format="valueFormat" @change="changeDate"
                  v-model="tempDate"></el-date-picker>
</template>

<script>
  import dayjs from "dayjs"

  export default {
    name: "DatePicker",
    props: {
      modelValue: {
        type: String,
        required: true
      },
      valueFormat: {
        type: String,
        default: function () {
          return "YYYY-MM-DD"
        }
      },
      placeholder: {
        type: String
      }
    },
    data() {
      return {
        tempDate: ""
      }
    },
    emits: ["update:modelValue", "change"],
    methods: {
      changeDate(data) {
        this.$emit("update:modelValue", dayjs(data).format(this.valueFormat))
        this.$emit("change", data)
      }
    },
    mounted() {
      if (this.modelValue) {
        this.tempDate = new Date(dayjs(this.modelValue).valueOf())
      }
    }
  }
</script>

```

**Comment by tracker1999** [comment]
@recursion123 
If you want encapsulate ElDatePicker into your own components

Use v-bind=$attrs to el-date-picker

<el-date-picker v-bind="$attrs" v-model="tempDate"></el-date-picker>

https://v3.vuejs.org/api/instance-properties.html#attrs

https://v3.vuejs.org/api/options-misc.html#inheritattrs

**Comment by iamkun** [comment]
fixed in #2120
